packages/back-end/src/validators/safe-rollout.ts

```diff
@@ -15,15 +15,32 @@ export type SafeRolloutStatus = typeof safeRolloutStatusArray[number];
 
 export const MaxDuration = z.object({
   amount: z.number(),
-  unit: z.enum(["days", "hours", "minutes"]),
+  unit: z.enum(["weeks", "days", "hours", "minutes"]),
 });
 // NB: These are the fields that the user submit, and the rest of the interface
 // is generated by us based on this information + feature & rule definition.
+const rampUpSchedule = z.object({
+  enabled: z.boolean(),
+  step: z.number(),
+  steps: z.array(
+    z.object({
+      percent: z.number(),
+      dateRampedUp: z.date(),
```

Comment:
question: Should this be `.optional()`? Or what value will this have for the steps that have not been ramped up yet?